hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -3366,3 +3376,64 @@ func (r *NodePoolReconciler) reconcileMirroredConfigs(ctx context.Context, logr
 	}
 	return nil
 }
+
+// SetPerformanceProfileConditions checks for performance profile status updates, and reflects them in the nodepool status conditions
+func (r *NodePoolReconciler) SetPerformanceProfileConditions(ctx context.Context, logger logr.Logger, nodePool *hyperv1.NodePool, controlPlaneNamespace string, toDelete bool) error {
+	if toDelete {
+		ppConds := []string{
+			hyperv1.NodePoolPerformanceProfileTuningAvailableConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningProgressingConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningUpgradeableConditionType,
+			hyperv1.NodePoolPerformanceProfileTuningDegradedConditionType,
+		}
+		for _, ppCond := range ppConds {
+			removeStatusCondition(&nodePool.Status.Conditions, ppCond)
+		}
+		return nil
+	}
+	// Get performance profile status configmap
+	cmList := &corev1.ConfigMapList{}
+	if err := r.Client.List(ctx, cmList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodeTuningGeneratedPerformanceProfileStatusLabel: "true"}),
+		Namespace:     controlPlaneNamespace,
+	}); err != nil {
+		return err
+	}
+	if len(cmList.Items) > 1 {
+		return fmt.Errorf("there cannot be more than one PerformanceProfile ConfigMap status per NodePool. found: %d NodePool: %s", len(cmList.Items), nodePool.Name)
+	}
+	if len(cmList.Items) == 0 {
+		// Only log here and do not return an error because it might take sometime for NTO to
```

Comment:
What happens if the CM takes forever to be created? (for whatever reason)

Reply:
@jparrill thank you for the thorough review. 
I don't think that a missing ConfigMap that represents the PerformanceProfile status should cause the controller to throw an error, which eventually requires user intervention to make the controller happy again. 
Printing a log should be a good compromisation  IMO.
Moreover, a missing CM might imply for a more severe issue in NTO which will eventually failed at some point to apply/read the PerfromanceProfile/Tuned objects. For such case Hypershift controller will throw an error.

That being said, I'm completely open to changes and I would appreciate your thought about the topic.

Reply:
I would log it as an `Error` instead of `Info` maybe not forcing the controller to fail and get blocked, but coming back returning `nil` and continuing with the reconcile loop until the user adds the needed things to make the operator to work normally. WDYT?

Reply:
@jparrill works for me. 
Once the ConfigMap gets created it will trigger the reconciliation loop.

Reply:
Done.